src/borg/archiver.py

```diff
@@ -2284,6 +2362,18 @@ def run(self, args):
             logger.warning("Using a pure-python msgpack! This will result in lower performance.")
         return args.func(args)
 
+    def _get_archives_slice(self, args, manifest):
+        if args.location.archive:
+            logger.error('The options --oldest and --latest must not be used on archive targets.')
+            self.exit_code = EXIT_ERROR
+            return []
+        n = args.oldest or args.latest
+        archives = manifest.list_archive_infos('ts', reverse=args.latest)[:n]
```

Comment:
maybe use reverse=bool(args.latest) - it is not strictly needed, but clearer that this will be evaluated as a boolean.